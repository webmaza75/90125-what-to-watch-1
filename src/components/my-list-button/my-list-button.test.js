import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import MyListButton from './my-list-button.jsx';

describe(`MyListButton correctly renders`, () => {
  it(`MyListButton correctly renders for a guest`, () => {
    const renderer = new ShallowRenderer();
    const result = renderer
      .render(<MyListButton
        isAuthorized={false}
        isFavorite={false}
        onMyListClick={jest.fn()}
        needRedirect={true}
      />);

    expect(result).toMatchSnapshot();
  });

  it(`MyListButton correctly renders for an authorized user`, () => {
    const renderer = new ShallowRenderer();
    const result = renderer
      .render(<MyListButton
        isAuthorized={true}
        isFavorite={false}
        onMyListClick={jest.fn()}
        needRedirect={false}
      />);

    expect(result).toMatchSnapshot();
  });
});
