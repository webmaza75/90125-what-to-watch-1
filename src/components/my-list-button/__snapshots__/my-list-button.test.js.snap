// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyListButton correctly renders MyListButton correctly renders for a guest 1`] = `
<Redirect
  to="/login"
/>
`;

exports[`MyListButton correctly renders MyListButton correctly renders for an authorized user 1`] = `
<button
  className="btn btn--list movie-card__button"
  onClick={[Function]}
  type="button"
>
  <svg
    height="20"
    viewBox="0 0 19 20"
    width="19"
  >
    <use
      xlinkHref="#add"
    />
  </svg>
  <span>
    My list
  </span>
</button>
`;
